lib/Doctrine/ORM/Mapping/DefaultEntityListenerResolver.php

```diff
@@ -58,7 +58,13 @@ public function register($object)
             throw new \InvalidArgumentException(sprintf('An object was expected, but got "%s".', gettype($object)));
         }
 
-        $this->instances[get_class($object)] = $object;
+        if ($object instanceof \ProxyManager\Proxy\VirtualProxyInterface) {
```

Comment:
Could something existing like `Doctrine\Common\Util\ClassUtil::getClass()` be used?